src/utils/createEndpoint.js

```diff
@@ -0,0 +1,58 @@
+import { HTTPSnippet } from 'httpsnippet';
+
+const supportedClients = {
+  curl: {
+    targetId: 'shell',
+    options: { short: true, indent: ' ', prettifyJson: true },
+  },
+};
+
+function validateData(data) {
+  if (typeof data !== 'object') {
+    throw new Error('A data object is required');
+  }
+
+  const requiredFields = [ 'method', 'path'];
+  requiredFields.forEach(field => {
+    if (!(field in data)) {
+      throw new Error(`Request is missing required field ${field}`);
+    }
+  });
+}
+
+function createSnippet(data) {
+  validateData(data);
```

Comment:
@MeganSteenkamp Suggestion: Perform validation before anything else (in the `createEndpoint` function)